src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -312,6 +312,13 @@ private TCommandResult ProcessResponse(ConnectionId connectionId, CommandMessage
                     if (materializedDocument.TryGetValue("errmsg", out errmsgBsonValue) && errmsgBsonValue.IsString)
                     {
                         var errmsg = errmsgBsonValue.ToString();
+                        if (materializedDocument.TryGetValue("code", out var errorCode)
```

Comment:
Maybe we can add a comment here why this logic is required?

Reply:
Should we create a ticket to wrap server error codes as ints into some class/enum? I don't really like this condition `== 20`. It doesn't say a lot :)

Reply:
Added a comment linking to the JIRA ticket. Also your suggestion sounds reasonable, go for it! Feel free to link back to this PR and this comment if needed.